<!DOCTYPE html>
<html>

  <head>
    <meta charset='utf-8' />
    <meta http-equiv="X-UA-Compatible" content="chrome=1" />
    <meta name="description" content="Multi-metagenome : " />

    <link rel="stylesheet" type="text/css" media="screen" href="stylesheets/stylesheet.css">

    <title>Multi-metagenome</title>
  </head>

  <body>

    <!-- HEADER -->
    <div id="header_wrap" class="outer">
        <header class="inner">
          <a id="forkme_banner" href="https://github.com/MadsAlbertsen/multi-metagenome">View on GitHub</a>

          <h1 id="project_title">Multi-metagenome</h1>
          <h2 id="project_tagline"></h2>

            <section id="downloads">
              <a class="zip_download_link" href="https://github.com/MadsAlbertsen/multi-metagenome/zipball/master">Download this project as a .zip file</a>
              <a class="tar_download_link" href="https://github.com/MadsAlbertsen/multi-metagenome/tarball/master">Download this project as a tar.gz file</a>
            </section>
        </header>
    </div>

    <!-- MAIN CONTENT -->
    <div id="main_content_wrap" class="outer">
      <section id="main_content" class="inner">
        <h1>Metagenome assembly guide</h1>

<p>This project contains scripts and tutorials on how to assemble individual microbial genomes from metagenomes, as described in Albertsen <em>et al</em>., 2013 (add link):</p>

<p><strong>Recovery of genomes from rare, uncultured bacteria by differential coverage binning of multiple deep metagenomes</strong></p>

<h2>Introduction</h2>

<p>Some text</p>

<h2>Data availability</h2>

<p>If you want to reanalyze the data used in this study the raw fastq reads can be obtained from the NCBI short read archive <a href="http://www.ncbi.nlm.nih.gov/sra/SRX206471?report=full">SRX206471 (HP+)</a> and <a href="http://www.ncbi.nlm.nih.gov/sra/SRX247688?report=full">SRX247688 (HP-)</a>. The assembled contigs can be obtained from NCBI GenBank under accesion number APMI01000000 (note: link not updated by NCBI yet). </p>

<p>In addition all processed data used in R can be obtained from HERE (add link).</p>

<h2>Step-by-step guide</h2>

<p>A detailed step-by-step guide is available in <a href="https://github.com/MadsAlbertsen/multi-metagenome">this folder</a>, which explains all steps from sampling to extraction of individual genomes.</p>

<p>A small shell script can be found in [R.data.generation] that wraps the individual scripts needed for generation of all data needed for subsequent genome extraction through R. The only input needed is a fasta file of the assembled scaffolds.</p>

<p>The R part of the step-by-step guide is also available as a seperate tutorial in R markdown format, which generates the main figures used in the article. Check it out HERE (add link).</p>

<p>05/2013 - Mads Albertsen - contact: ma at bio dot aau dot dk</p>
      </section>
    </div>

    <!-- FOOTER  -->
    <div id="footer_wrap" class="outer">
      <footer class="inner">
        <p class="copyright">Multi-metagenome maintained by <a href="https://github.com/MadsAlbertsen">MadsAlbertsen</a></p>
        <p>Published with <a href="http://pages.github.com">GitHub Pages</a></p>
      </footer>
    </div>

    

  </body>
</html>
