<!DOCTYPE html>
<html>
  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <link href='https://fonts.googleapis.com/css?family=Architects+Daughter' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="http://madsalbertsen.github.io/multi-metagenome/css/stylesheet.css" media="screen" />
    <title>Multi-metagenome</title>
  </head>


  <body>
<header>
  <div class="inner">
    <a id="forkme_banner" href="https://github.com/MadsAlbertsen/multi-metagenome">View on GitHub</a>
    <h1>
      <a style="color: rgb(255,255,255)" href="/multi-metagenome"> 
      Multi-metagenome
    </h1>
    <h2>Recovery of complete genomes from metagenomes</h2>
    <section id="downloads">
      <a class="zip_download_link" href="https://github.com/MadsAlbertsen/multi-metagenome/zipball/master">Download this project as a .zip file</a>
      <a class="tar_download_link" href="https://github.com/MadsAlbertsen/multi-metagenome/tarball/master">Download this project as a tar.gz file</a>
    </section>
  </div>
</header>


    <div id="content-wrapper">
      <div class="inner clearfix">
        <section id="main-content">
          <h1 id='under_construction_'>Under construction !</h1>

<h2 id='finishing'>Finishing</h2>

<p>The last part of the guide is the finishing step. The extracted genome is by now probably already great quality. However, to make it into a high-quality genome it is needed to dive into the genome at nucleotide resolution, while still keeping an overview of the full genome.</p>

<p>To facilitate this we have integrated all relevant data in the framework of <a href='http://circos.ca/'>Circos</a>. Circos enables circular visualisations of any type of data, but is often used for genomic data. Circos have a steep learning curve, but also one of the best online guides we have seen so far (<a href='http://circos.ca/documentation/tutorials/'>check it out</a>). After trying out the scripts in this section you should be able to generate a plot as the one shown below for any genome assembly.</p>

<p><img alt='Circos initial' src='figure/circos1.png' /></p>

<h3 id='generating_data_for_visualisation_in_circos'>Generating data for visualisation in Circos</h3>

<p>The script <code>circosviz.pl</code> generates all data needed for the Circos visualisation. The only needed input is a SAM file of the reads mapped to the assembly and a fasta file of the assembled scaffolds.</p>
<div class='highlight'><pre><code class='text'>perl multi-metagenome/circosviz/circosviz.pl -i mapping.sam -f scaffolds.fa -e 500 -m 3000 -a 125 -b 1000 -p 1000
</code></pre></div>
<p><code>-i</code>: A SAM file of the reads mapped to the scaffolds. The requirement to the read header is the same as in cytoscapeviz.pl (see <a href='step10.html'>PE tracking</a>).</p>

<p><code>-f</code>: A fasta file of the assembled scaffolds. This is used to calculate GC profiles and mark N&#8217;s.</p>

<p><code>-e</code>: The maximum distance from the start or end of a scaffold for a read to be considered an end read.</p>

<p><code>-m</code>: The minimum length for a scaffold to allow for end-end connections.</p>

<p><code>-a</code>: The average read length of the reads mapped to the scaffolds.</p>

<p><code>-b</code>: The bin size for coverage, GC calculations and PE mapping density calculations.</p>

<p><code>-p</code>: The minumum distance between reads to be considered a split PE read. Only spilt PE reads are visualised.</p>

<p><code>-g</code>: Adds <a href='http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0052210'>FRCbam</a> assembly metrics using the Feature.gff output of the program.</p>

<h4 id='output_karyotype'>Output: Karyotype</h4>

<p>The basic output file is <code>circos.karyotype.txt</code>, which contains the names and lengths of the scaffolds in the assembly. The scaffolds are arranged by decreasing length and assigned a color. This <code>.karyotype.</code> file is used by Circos as a backbone for the plot. Below is an example of the file provided in the example dataset (see <code>circosviz/circosexample/data</code>). Each scaffold is represented as a chromosome using the <code>chr</code>-tag and any gaps in the scaffolds are marked using the <code>band</code>-tag.</p>
<div class='highlight'><pre><code class='text'>chr - scaffold1 scaffold1 0 206839 set1-7-qual-1
...
band scaffold1 band1 band1 19877 19957 black
</code></pre></div>
<p><code>set1-7-qual-1</code> refer to a specific color from the <a href='http://www.colorbrewer2.org/'>Brewer palettes</a> which is integrated in Circos.</p>

<h4 id='output_coverage_and_gc_content'>Output: Coverage and GC content</h4>

<p>The <code>circos.coverage.txt</code> file contians the coverage of each scaffold in a given bin size (<code>-b</code>). One of the nice things by Circos is the simple syntax for adding new data to the plot. Here, a coverage of <code>396.1</code> is assigned to the first 1000 bp of <code>scaffold1</code>.</p>
<div class='highlight'><pre><code class='text'>scaffold1 0 1000 396.1
</code></pre></div>
<p>The file <code>circos.gc.txt</code> contains the GC content of all scaffolds in a given bin size (<code>-b</code>) and have a similar structure as the <code>.coverage.</code> file.</p>

<h4 id='output_pairedend_links'>Output: Paired-end links</h4>

<p>A central feature of the assembly visulisation is the ability to track where all paired-end (PE) reads map in the assembly. The paired-end reads are classified into four groups.</p>

<p><img alt='Circos PE classes' src='figure/circosClass.png' /></p>

<p>The first group is <code>good PE</code> reads which are defined as having an insert size within a defined cutoff (<code>-p</code>). The second group is <code>good end</code> links. These are defined as PE reads where both reads map within a given distance (<code>-e</code>) of either the start or end of the scaffold. The <code>good end</code> links represent ends of scaffolds that might be manually scaffolded.</p>

<p>The third group is <code>bad between</code> links, where the PE reads map to different scaffolds, but at least one of the PE reads do not map within an end. The <code>bad between</code> links can indicate repeat integration in the genome or chimeric regions. Lastly there is the <code>bad within</code> group, which are PE reads that map within the same scaffold and have an insert size above the defined cutoff (<code>-p</code>).</p>

<p>Only the <code>good end</code>, <code>bad between</code> and <code>bad within</code> links are reported. The associated files are named <code>circos.ends.txt</code>, <code>circos.dcontigs.txt</code> and <code>circos.scontigs.wl.txt</code> respectively. The file structure for links in Circos is shown below. Each PE read is assigned a group, e.g. <code>pe1</code> and the read mapping position is assigned.</p>
<div class='highlight'><pre><code class='text'>pe1 scaffold1     100    101
pe1 scaffold2  206676 206677
pe2 scaffold7      11     12
pe2 scaffold11 100234 100235
</code></pre></div>
<h4 id='output_link_density'>Output: Link density</h4>

<p>In most assemblies there will be millons of reads, which will result in a lot of false-positives. In order to determine if a region is problematic the density of reads in the different link groups is reported for a given bin size <code>-b</code>. These files are denoted with <code>.count.</code>. The link density files have the same structure as the <code>coverage</code> file. E.g. there is <code>173</code> <code>good end</code> links in the first 500 bp of <code>scaffold1</code>.</p>
<div class='highlight'><pre><code class='text'>scaffold1 0 500 173
</code></pre></div>
<h4 id='output_frcbam_statistics'>Output: FRCbam statistics</h4>

<p>The file <code>frx.tracks.txt</code> contains <a href='http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0052210'>FRCbam</a> reference free validation statistics that can often be usefull to identify problematic regions in assemblies. In order to comply with Circos the different statistics have been given numbers (see later). E.g. <code>4</code> represents <code>LOW_COV_PE</code> which stands for low paired-end coverage.</p>
<div class='highlight'><pre><code class='text'>scaffold1 3025 3300 4
</code></pre></div>
<h3 id='generating_circos_figures'>Generating Circos figures</h3>

<p>Explain what the Circos configuration file is.</p>
<div class='highlight'><pre><code class='text'>circos -conf etc\denovoviz_1_clean.conf -noparanoid
</code></pre></div>
<p><img alt='Circos initial' src='figure/circos1.png' /></p>

<h3 id='manipulating_the_circos_figures'>Manipulating the Circos figures</h3>

<p>Show how to add the different kinds of information: First all data no filtering at all.</p>

<h3 id='curating_the_assembly'>Curating the assembly</h3>

<p>Explain how to inspect and correct the assembly based on the Circos figures. Zoom etc.</p>

<p>Finishing is by far the most time-consuming step and often a reasonable draft genome is all that is needed. In this case use the Circos visual&#8230;</p>

<h3 id='tweaks_and_tricks'>Tweaks and tricks</h3>

<p>Extract the part of a SAM file that is associated with a given list of scaffolds.</p>
        </section>
        <aside id="sidebar">
  <p>
    <a href="/multi-metagenome/" class="button">Home</a>
  </p>
  <p class="">
    <a href="/multi-metagenome/docs/overview.html" class="button">Step-by-step guide</a>
  </p>
  <p class=""> 
    <a href="/multi-metagenome/docs/step1.html" class="button">-- Samples</a>
  </p>
  <p class=""> 
    <a href="/multi-metagenome/docs/step3.html" class="button">-- Assembly</a>
  </p>
  <p class=""> 
    <a href="/multi-metagenome/docs/step5.html" class="button">-- Data generation</a>
  </p>
  <p class=""> 
    <a href="/multi-metagenome/docs/step9.html" class="button">-- Binning</a>
  </p>
  <p class=""> 
    <a href="/multi-metagenome/docs/step10.html" class="button">-- PE tracking</a>
  </p>
  <p class=""> 
    <a href="/multi-metagenome/docs/step11.html" class="button">-- Reassembly</a>
  </p>
  <p class="current">
    <a href="/multi-metagenome/docs/step12.html" class="button">-- Finishing</a>
  </p>
  <p class=""> 
    <a href="/multi-metagenome/docs/faq.html" class="button">FAQ</a>
  </p>
</aside>

      </div>
    </div>

<footer>
  <div class="inner">
    <p>
      Multi-metagenome maintained by 
      <a href="https://github.com/MadsAlbertsen">MadsAlbertsen</a>
    </p>
    <p>Published with <a href="http://pages.github.com">GitHub Pages</a> using <a href="http://jekyllrb.com/">Jekyll</a>
    </p>
  </div>
</footer>

<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-41280485-1']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

  </body>
</html>



