<!DOCTYPE html>
<html>
  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <link href='https://fonts.googleapis.com/css?family=Architects+Daughter' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="/css/stylesheet.css" media="screen" />
    <title>Multi-metagenome</title>
  </head>


  <body>
<header>
  <div class="inner">
    <a id="forkme_banner" href="https://github.com/MadsAlbertsen/multi-metagenome">View on GitHub</a>
    <h1>
      <a style="color: rgb(255,255,255)" href="/"> 
      Multi-metagenome
    </h1>
    <h2>Recovery of complete genomes from metagenomes</h2>
    <section id="downloads">
      <a class="zip_download_link" href="https://github.com/MadsAlbertsen/multi-metagenome/zipball/master">Download this project as a .zip file</a>
      <a class="tar_download_link" href="https://github.com/MadsAlbertsen/multi-metagenome/tarball/master">Download this project as a tar.gz file</a>
    </section>
  </div>
</header>


    <div id="content-wrapper">
      <div class="inner clearfix">
        <section id="main-content">
          <h2 id='data_generation'>Data generation</h2>

<p>This section describes how to generate all data needed for subsequent binning in R. The data generation is diveded into a few small scripts in order to allow flexiblity in the choice of tools used. The important part is not which tools are used, but that all the data is integrated in R subsequently.</p>

<p>For each scaffold the following information is generated: <code>gc content</code>, <code>kmer frequency</code>, <code>essential genes</code>, <code>essential genes taxonomy</code>. The full processed dataset used in the paper can be obtained from <a href='https://dl.dropbox.com/s/989dix16ugyuvrq/Albertsen2013.data.tar.gz'>here</a>.</p>

<h3 id='dependencies'>Dependencies</h3>

<p>A few programs are used in the data generation and hence needs to be installed: <a href='http://bioinformatics.oxfordjournals.org/content/28/17/2223.abstract'>MetaProdigal</a>, <a href='http://hmmer.janelia.org/'>HMMER 3.0</a>, <a href='http://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download'>BLAST</a> and <a href='http://ab.inf.uni-tuebingen.de/software/megan/'>MEGAN</a>.</p>

<h2 id='for_the_impatient'>For the impatient</h2>

<p>Download the multi-metagenome scripts. The easy way is to clone the github repository:</p>
<div class='highlight'><pre><code class='text'>git clone git://github.com/MadsAlbertsen/multi-metagenome.git
</code></pre></div>
<p>The shell script <code>workflow.R.data.generation.sh</code> in the <strong>R.data.generation</strong> folder wraps the individual steps. It assumes that the de novo assembly is named assembly.fa and present in the same folder as the script. Assuming you are in the same folder as where you downloaded the <strong>multi-metagenome</strong> folder, copy the shell script and then run it:</p>
<div class='highlight'><pre><code class='text'>cp multi-metagenome/R.data.generation/workflow.R.data.generation.sh .
sh workflow.R.data.generation.sh
</code></pre></div>
<p>The following files are generated: <code>assembly.gc.tab</code>, <code>assembly.kmer.tab</code>, <code>assembly.orfs.hmm.id.txt</code>,<code>assembly.orfs.hmm.blast.tax.tab</code>, <code>assembly.tax.consensus.txt</code>.</p>

<h2 id='data_generation_in_details'>Data generation in details</h2>

<p>Below the individual steps in the data generation process is described. The full path to the scripts are listed to show where they are located in the <strong>multi-metagenome</strong> folder.</p>

<p>Note: several of the scripts and programs rely on the header structure of the input sequences, to avoid any potential problems rename the scaffolds as e.g. 1, 2, 3 &#8230; 100110 etc.</p>

<h3 id='scaffold_coverage_and_length'>Scaffold coverage and length</h3>

<p>Scaffold coverage is estimated by mapping the reads from the two samples independently to the assembled scaffolds. Again we use CLC, but any other open source tool could be used. The result of this step is a list of all scaffolds and their coverage in either sample. In CLC the data is exported as a simple <code>.csv</code> file which contains both the <code>coverage</code> and <code>length</code> of each scaffold. The files <code>HPminus.scaffold.coverage.csv</code> and <code>HPplus.scaffold.coverage.csv</code> are included in the <a href='https://dl.dropbox.com/s/989dix16ugyuvrq/Albertsen2013.data.tar.gz'>example dataset</a>.</p>

<h3 id='tetranucleotide_frequency'>Tetranucleotide frequency</h3>

<p>Tetranucleotide frequencies patterns are generated from the assembled scaffolds using <code>calc.kmerfreq.pl</code>.</p>
<div class='highlight'><pre><code class='text'>perl multi-metagenome/R.data.generation/calc.kmerfreq.pl -i assembly.fa -o assembly.kmer.tab
</code></pre></div>
<h3 id='gc_content'>GC content</h3>

<p>The GC content of each scaffold is calculated using <code>calc.gc.pl</code>.</p>
<div class='highlight'><pre><code class='text'>perl multi-metagenome/R.data.generation/calc.gc.pl -i assembly.fa -o assembly.gc.tab
</code></pre></div>
<h3 id='identification_of_conserved_marker_proteins'>Identification of conserved marker proteins</h3>

<p>A set of 100+ HMM models (<code>essential.hmm</code>) of conserved single copy marker proteins is used to evaluate completeness and contamination in the genome bins and also to guide the initial selection of genome bins.</p>

<p>As the HMM models are on protein level it is initially needed to call open reading frames in the assembled scaffolds using <a href='http://bioinformatics.oxfordjournals.org/content/28/17/2223.abstract'>MetaProdigal</a>. The second line just simplifies the header a little.</p>
<div class='highlight'><pre><code class='text'>prodigal -a temp.orfs.faa -i assembly.fa -m -o temp.txt -p meta -q
cut -f1 -f “ “ temp.orfs.faa &gt; assembly.orfs.faa
</code></pre></div>
<p>Essential proteins are identified using HMM models (<code>essential.hmm</code>) and the output reformatted for later use in R. The file <code>assembly.orfs.hmm.id.txt</code> is used later in R.</p>
<div class='highlight'><pre><code class='text'>hmmsearch --tblout assembly.hmm.orfs.txt --cut_tc --notextw multi-metagenome/R.data.generation/essential.hmm assembly.orfs.faa
tail -n+4 assembly.hmm.orfs.txt | sed &#39;s/ * / /g&#39; | cut -f1,4 -d &quot; &quot; &gt; assembly.orfs.hmm.id.txt
</code></pre></div>
<p>The file <code>assembly.hmm.orfs.txt</code> contains information of all HMM positive ORFs. In order to get a fast overview of their taxonomic affiliation the names of the HMM positive ORFs are extracted (<code>list.of.positive.orfs.txt</code>) the sequence of the positive ORFs are extracted using <code>extract.using.header.list.pl</code> and then searched against the NCBI refseq_protein database using <a href='http://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download'>BLAST</a>.</p>
<div class='highlight'><pre><code class='text'>tail -n+4 assembly.hmm.orfs.txt | cut -f1 -d &quot; &quot; &gt; list.of.positive.orfs.txt
perl multi-metagenome/R.data.generation/extract.using.header.list.pl -l list.of.positive.orfs.txt -s assembly.orfs.faa -o assembly.orfs.hmm.faa
blastp -query assembly.orfs.hmm.faa -db refseq_protein -evalue 1e-5 -num_threads 60 -max_target_seqs 5 -outfmt 5 -out assembly.orfs.hmm.blast.xml
</code></pre></div>
<p>To get a reasonable taxonomic classification <a href='http://ab.inf.uni-tuebingen.de/software/megan/'>MEGANs</a> LCA algorithm is utilized and the taxonomic assignment of each protein exported and formated for later use in R. The file <code>assembly.orfs.hmm.blast.tax.tab</code> contains the taxonomic assignment for each essential gene.</p>
<div class='highlight'><pre><code class='text'>MEGAN +g -x &quot;import blastfile= assembly.orfs.hmm.blast.xml meganfile=temp.rma;recompute toppercent=5;recompute minsupport=1;update;collapse rank=Species;update;select nodes=all;export what=CSV format=readname_taxonpath separator=tab file=assembly.orfs.hmm.blast.tax.txt;update;close&quot;
sed &#39;s/\t/;/&#39; assembly.orfs.hmm.blast.tax.txt | cut -f1,5 -d &quot;;&quot; | sed &#39;s/;/\t/&#39; | sed &#39;s/_/\t/&#39; &gt; assembly.orfs.hmm.blast.tax.tab
</code></pre></div>
<p>As the marker proteins are also used for guiding the binning process it is needed to aggregate the protein results to scaffold level. However, as large scaffolds often contain multiple marker proteins the script <code>hmm.majority.vote.pl</code> is used to find the consensus taxonomic assignment on scaffold level (<code>assembly.tax.consensus.txt</code>).</p>
<div class='highlight'><pre><code class='text'>perl multi-metagenome/R.data.generation/hmm.majority.vote.pl -i assembly.orfs.hmm.blast.tax.txt -o assembly.tax.consensus.txt -n
</code></pre></div>
<p><a href='/docs/step9.html'>Next: Binning using R</a></p>
        </section>
        <aside id="sidebar">
  <p>
    <a href="/" class="button">Home</a>
  </p>
  <p class="">
    <a href="/docs/overview.html/" class="button">Step-by-step guide</a>
  </p>
  <p class=""> 
    <a href="/docs/step1.html/" class="button">-- Samples</a>
  </p>
  <p class=""> 
    <a href="/docs/step3.html/" class="button">-- Assembly</a>
  </p>
  <p class="current"> 
    <a href="/docs/step5.html/" class="button">-- Data generation</a>
  </p>
  <p class=""> 
    <a href="/docs/step9.html/" class="button">-- Binning</a>
  </p>
  <p class=""> 
    <a href="/docs/step10.html/" class="button">-- PE tracking</a>
  </p>
  <p class=""> 
    <a href="/docs/step11.html/" class="button">-- Reassembly</a>
  </p>
  <p class="">
    <a href="/docs/step12.html/" class="button">-- Finishing</a>
  </p>
  <p class=""> 
    <a href="/docs/faq.html/" class="button">FAQ</a>
  </p>
  <p class=""> 
    <a href="/docs/contact.html/" class="button">Contact</a>
  </p>
</aside>

      </div>
    </div>

<footer>
  <div class="inner">
    <p>
      Multi-metagenome maintained by 
      <a href="https://github.com/MadsAlbertsen">MadsAlbertsen</a>
    </p>
    <p>Published with <a href="http://pages.github.com">GitHub Pages</a> using <a href="http://jekyllrb.com/">Jekyll</a>
    </p>
  </div>
</footer>


  </body>
</html>



