<!DOCTYPE html>
<html>
  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <link href='https://fonts.googleapis.com/css?family=Architects+Daughter' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="/css/stylesheet.css" media="screen" />
    <title>Multi-metagenome</title>
  </head>


  <body>
<header>
  <div class="inner">
    <a id="forkme_banner" href="https://github.com/MadsAlbertsen/multi-metagenome">View on GitHub</a>
    <h1>
      <a style="color: rgb(255,255,255)" href="/"> 
      Multi-metagenome
    </h1>
    <h2>Recovery of complete genomes from metagenomes</h2>
    <section id="downloads">
      <a class="zip_download_link" href="https://github.com/MadsAlbertsen/multi-metagenome/zipball/master">Download this project as a .zip file</a>
      <a class="tar_download_link" href="https://github.com/MadsAlbertsen/multi-metagenome/tarball/master">Download this project as a tar.gz file</a>
    </section>
  </div>
</header>


    <div id="content-wrapper">
      <div class="inner clearfix">
        <section id="main-content">
          <h2 id='which_samples_can_be_used'>Which samples can be used?</h2>

<p>At least two samples is needed where the target organisms are in different abundances. In the paper we applied two different extraction methods to the same sludge sample, which resulted in different abundances of the target orgamisms, due to species specific extraction properties. However, we have also succesfully used timeseries data, and spatial related data. Any combination of samples can be used as long as you have the target organisms in both samples.</p>

<h3 id='microdiversity'>Micro-diversity</h3>

<p>The approach only concerns the binning, finishing and validation part of assembling genomes from metagenomes. If there are many strains of the same species present in the sample they will <strong>not</strong> be assembled in the de novo assembly. Hence, the approach can help you to get the most out of what you could assemble. It can not help you to recover complete genomes in samples with massive strain diversity of the target species as they will not assemble in the first place.</p>

<p><a href='/docs/step3.html'>Next: Sequencing and Assembly</a></p>
        </section>
        <aside id="sidebar">
  <p>
    <a href="/" class="button">Home</a>
  </p>
  <p class="">
    <a href="/docs/overview.html/" class="button">Step-by-step guide</a>
  </p>
  <p class="current"> 
    <a href="/docs/step1.html/" class="button">-- Samples</a>
  </p>
  <p class=""> 
    <a href="/docs/step3.html/" class="button">-- Assembly</a>
  </p>
  <p class=""> 
    <a href="/docs/step5.html/" class="button">-- Data generation</a>
  </p>
  <p class=""> 
    <a href="/docs/step9.html/" class="button">-- Binning</a>
  </p>
  <p class=""> 
    <a href="/docs/step10.html/" class="button">-- PE tracking</a>
  </p>
  <p class=""> 
    <a href="/docs/step11.html/" class="button">-- Reassembly</a>
  </p>
  <p class="">
    <a href="/docs/step12.html/" class="button">-- Finishing</a>
  </p>
  <p class=""> 
    <a href="/docs/faq.html/" class="button">FAQ</a>
  </p>
  <p class=""> 
    <a href="/docs/contact.html/" class="button">Contact</a>
  </p>
</aside>

      </div>
    </div>

<footer>
  <div class="inner">
    <p>
      Multi-metagenome maintained by 
      <a href="https://github.com/MadsAlbertsen">MadsAlbertsen</a>
    </p>
    <p>Published with <a href="http://pages.github.com">GitHub Pages</a> using <a href="http://jekyllrb.com/">Jekyll</a>
    </p>
  </div>
</footer>


  </body>
</html>



