<!DOCTYPE html>
<html>
  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <link href='https://fonts.googleapis.com/css?family=Architects+Daughter' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="http://madsalbertsen.github.io/multi-metagenome/css/stylesheet.css" media="screen" />
    <title>Multi-metagenome</title>
  </head>


  <body>
<header>
  <div class="inner">
    <a id="forkme_banner" href="https://github.com/MadsAlbertsen/multi-metagenome">View on GitHub</a>
    <h1>
      <a style="color: rgb(255,255,255)" href="/multi-metagenome"> 
      Multi-metagenome
    </h1>
    <h2>Recovery of complete genomes from metagenomes</h2>
    <section id="downloads">
      <a class="zip_download_link" href="https://github.com/MadsAlbertsen/multi-metagenome/zipball/master">Download this project as a .zip file</a>
      <a class="tar_download_link" href="https://github.com/MadsAlbertsen/multi-metagenome/tarball/master">Download this project as a tar.gz file</a>
    </section>
  </div>
</header>


    <div id="content-wrapper">
      <div class="inner clearfix">
        <section id="main-content">
          <p>This project contains scripts and tutorials on how to assemble individual microbial genomes from metagenomes, as described in:</p>

<p><strong>Genome sequences of rare, uncultured bacteria obtained by differential coverage binning of multiple metagenomes</strong></p>

<p><a href='http://personprofil.aau.dk/120257'>Mads Albertsen</a>, <a href='http://ecogenomic.org/users/phil-hugenholtz'>Philip Hugenholtz</a>, <a href='http://ecogenomic.org/users/adam-skarshewski'>Adam Skarshewski</a>, <a href='http://www.ecogenomic.org/users/gene-tyson'>Gene W. Tyson</a>, <a href='http://personprofil.aau.dk/103057'>KÃ¥re L. Nielsen</a> and <a href='http://personprofil.aau.dk/105842'>Per .H. Nielsen</a></p>

<p>Nature Biotechnology 2013, doi: <a href='http://www.nature.com/nbt/journal/vaop/ncurrent/abs/nbt.2579.html'>10.1038/nbt.2579</a></p>

<h2 id='what_is_differential_coverage_binning'>What is differential coverage binning?</h2>

<p>Differential coverage binning refers to the use of the abundance of the bacteria in the samples as the primary method of extracting genomes from metagenomes. This is compared to many other binning methods where sequence composition (e.g. tetranucleotide patterns) is used for binning, which is hampered by local sequence deviations within genomes.</p>

<p>The great advantage of using abundance is that it allows the use of abundance estimates from multiple samples, thereby increasing the binning resolution greatly. Given that sequencing prices continiues to drop it is already cheaper to generate data than to analyse it.</p>

<h2 id='stepbystep_guide'>Step-by-step guide</h2>

<p>The <a href='docs/overview.html'>guide</a> covers all aspects, from which samples that can be used, to binning, finishing and validation of the extracted genomes. <a href='http://www.rstudio.com/'>Rstudio</a> (a powerfull IDE to <a href='http://www.r-project.org/'>R</a>) is used as the main tool for data handling as it allows integration of all relevant data, which is key for dealing with large and complex datasets. The guide to binning in R is also available in R markdown format <a href='https://github.com/MadsAlbertsen/multi-metagenome/tree/master/R.markdown.guide'>here</a>, which allows direct recreation of all figures in the <a href='docs/overview.html'>guide</a> using the original data from the paper.</p>

<p>In the <a href='docs/overview.html'>overview</a> section a short description of the workflow is given, along with a detailed workflow figure that summarise the different steps in the process of assembling individual genomes from metagenomes.</p>

<p>In addition to the online guide a PDF version of the original published guide is available <a href='https://github.com/MadsAlbertsen/multi-metagenome'>here</a>.</p>

<h2 id='data_availability'>Data availability</h2>

<p>If you want to reanalyze the data used in this study, the raw fastq reads can be obtained from NCBI SRA: <a href='http://www.ncbi.nlm.nih.gov/sra/SRX206471?report=full'>SRX206471 (HP+)</a> and <a href='http://www.ncbi.nlm.nih.gov/sra/SRX247688?report=full'>SRX247688 (HP-)</a>. The assembled contigs can be obtained from NCBI GenBank under accesion number <a href='http://www.ncbi.nlm.nih.gov/nuccore/494587257'>APMI01000000</a>.</p>

<p>In addition, all processed data ready for binning in R can be obtained <a href='https://dl.dropbox.com/s/989dix16ugyuvrq/Albertsen2013.data.tar.gz'>here</a>.</p>
        </section>
        <aside id="sidebar">
  <p>
    <a href="/multi-metagenome/" class="button">Home</a>
  </p>
  <p class="">
    <a href="/multi-metagenome/docs/overview.html" class="button">Step-by-step guide</a>
  </p>
  <p class=""> 
    <a href="/multi-metagenome/docs/step1.html" class="button">-- Samples</a>
  </p>
  <p class=""> 
    <a href="/multi-metagenome/docs/step3.html" class="button">-- Assembly</a>
  </p>
  <p class=""> 
    <a href="/multi-metagenome/docs/step5.html" class="button">-- Data generation</a>
  </p>
  <p class=""> 
    <a href="/multi-metagenome/docs/step9.html" class="button">-- Binning</a>
  </p>
  <p class=""> 
    <a href="/multi-metagenome/docs/step10.html" class="button">-- PE tracking</a>
  </p>
  <p class=""> 
    <a href="/multi-metagenome/docs/step11.html" class="button">-- Reassembly</a>
  </p>
  <p class="">
    <a href="/multi-metagenome/docs/step12.html" class="button">-- Finishing</a>
  </p>
  <p class=""> 
    <a href="/multi-metagenome/docs/faq.html" class="button">FAQ</a>
  </p>
</aside>

      </div>
    </div>

<footer>
  <div class="inner">
    <p>
      Multi-metagenome maintained by 
      <a href="https://github.com/MadsAlbertsen">MadsAlbertsen</a>
    </p>
    <p>Published with <a href="http://pages.github.com">GitHub Pages</a> using <a href="http://jekyllrb.com/">Jekyll</a>
    </p>
  </div>
</footer>


  </body>
</html>



